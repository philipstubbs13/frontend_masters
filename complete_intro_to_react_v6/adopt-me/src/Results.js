import Pet from "./Pet";

const Results = ({ pets }) => {
  return (
    <div className="search">
      {!pets.length ? (
        <h2>No Pets Found</h2>
      ) : (
        pets.map((pet) => (
          <Pet
            animal={pet.animal}
            key={pet.id}
            name={pet.name}
            breed={pet.breed}
            images={pet.images}
            location={`${pet.city}, ${pet.state}`}
            id={pet.id}
          />
        ))
      )}
    </div>
  );
};

export default Results;
