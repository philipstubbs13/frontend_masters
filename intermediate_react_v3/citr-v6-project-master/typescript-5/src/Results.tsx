import { FunctionComponent } from "react";
import { Pet as PetType } from "./APIResponsesTypes";
import Pet from "./Pet";

const Results: FunctionComponent<{ pets: PetType[] }> = ({ pets }) => {
  return (
    <div className="search">
      {!pets.length ? (
        <h1>No Pets Found</h1>
      ) : (
        pets.map((pet) => {
          return (
            <Pet
              animal={pet.animal}
              key={pet.id}
              name={pet.name}
              breed={pet.breed}
              images={pet.images}
              location={`${pet.city}, ${pet.state}`}
              id={pet.id}
            />
          );
        })
      )}
    </div>
  );
};

export default Results;
