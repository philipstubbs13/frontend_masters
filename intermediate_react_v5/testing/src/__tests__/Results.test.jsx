import { expect, test } from 'vitest';
import { render } from '@testing-library/react';
import Results from '../Results';

test("renders correct with no pets", async () => {
    const { asFragment } = render(<Results pets={[]} />)
    expect(asFragment()).toMatchSnapshot();
})