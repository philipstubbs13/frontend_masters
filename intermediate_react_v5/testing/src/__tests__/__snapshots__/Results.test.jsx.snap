// Vitest Snapshot v1

exports[`renders correct with no pets 1`] = `
<DocumentFragment>
  <div
    class="search"
  >
    <h1>
      No Pets Found
    </h1>
  </div>
</DocumentFragment>
`;
